### ISSUE TITLE
Unpinned dependency on graphql-core leads to issues

### ISSUE THREAD

**Issue by ryanwilsonperkin** [graphql-python/graphene#580]
We use `graphene` 1.4.1 and haven't yet updated to the recently released 2.0. Because the setup.py [only specifies a minimum version for `graphql-core`](https://github.com/graphql-python/graphene/blob/v1.4.1/setup.py#L85), a recent deploy pulled in `graphql-core` 2.0 with `graphene` 1.4.1, versions which are incompatible and resulted in breaking changes.

This is related to a similar issue raised in `graphene-django` (https://github.com/graphql-python/graphene-django/issues/277).

Would you be willing to cut a 1.4.2 release that pinned `graphql-core` to `<2`?

**Comment by ryanwilsonperkin** [comment]
Here's an example of an error:
```python
GraphQLLocatedError: resolve() got an unexpected keyword argument 'id'
  File "graphql/execution/executors/utils.py", line 6, in process
    val = f(*args, **kwargs)
```

**Comment by syrusakbary** [comment]
Just published a new version of Graphene `1.4.2` with the pinned requirements.
https://pypi.python.org/pypi/graphene/1.4.2

Closing the PR :)

**Comment by ryanwilsonperkin** [comment]
Thanks Syrus, you're the best. Congrats on 2.0!
